test/common/common.go

```diff
@@ -0,0 +1,483 @@
+	operatorServiceAccount     = "external-dns-operator"
+	OperatorNamespace          = "external-dns-operator"
+	dnsProviderEnvVar          = "DNS_PROVIDER"
+	e2eSkipDNSProvidersEnvVar  = "E2E_SKIP_DNS_PROVIDERS"
+	DnsPollingInterval         = 15 * time.Second
+	DnsPollingTimeout          = 3 * time.Minute
+)
+
+var (
+	KubeClient    client.Client
+	KubeClientSet *kubernetes.Clientset
+)
+
+func init() {
+	var err error
+	if err, KubeClient, KubeClientSet = initKubeClient(); err != nil {
+		fmt.Printf("Failed to init kube client: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func DefaultService(name, namespace string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"external-dns.mydomain.org/publish": "yes",
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"name": "hello-openshift",
+			},
+			Type: corev1.ServiceTypeLoadBalancer,
+			Ports: []corev1.ServicePort{
+				{
+					Protocol: corev1.ProtocolTCP,
+					Port:     80,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 8080,
+					},
+				},
+			},
+		},
+	}
+}
+
+func MustGetEnv(name string) string {
+	val := os.Getenv(name)
+	if val == "" {
+		panic(fmt.Sprintf("environment variable %s must be set", name))
+	}
+	return val
+}
+
+func RootCredentials(kubeClient client.Client, name string) (map[string][]byte, error) {
+	secret := &corev1.Secret{}
+	secretName := types.NamespacedName{
+		Name:      name,
+		Namespace: "kube-system",
+	}
+	if err := kubeClient.Get(context.TODO(), secretName, secret); err != nil {
+		return nil, fmt.Errorf("failed to get credentials secret %s: %w", secretName.Name, err)
+	}
+	return secret.Data, nil
+}
+
+func LookupARecord(host, server string) ([]string, error) {
+	dnsClient := &dns.Client{}
+	message := &dns.Msg{}
+	message.SetQuestion(dns.Fqdn(host), dns.TypeA)
+	response, _, err := dnsClient.Exchange(message, fmt.Sprintf("%s:53", server))
+	if err != nil {
+		return nil, err
+	}
+	if len(response.Answer) == 0 {
+		return nil, fmt.Errorf("not found")
+	}
+	var ips []string
+	for _, ans := range response.Answer {
+		if aRec, ok := ans.(*dns.A); ok {
+			ips = append(ips, aRec.A.String())
+		}
+	}
+	if len(ips) == 0 {
+		return nil, fmt.Errorf("not found")
+	}
+	return ips, nil
+}
+
+// LookupARecordInternal queries for a DNS hostname inside the VPC of a cluster using a dig pod. It returns a
+// map structure of the resolved IPs for easy lookup.
+func LookupARecordInternal(ctx context.Context, t *testing.T, cl client.Client, cs *kubernetes.Clientset, namespace, host string) (map[string]struct{}, error) {
+	t.Helper()
+
+	clientPod := buildDigPod("digpod", namespace, host)
+	if err := cl.Create(ctx, clientPod); err != nil {
+		return nil, fmt.Errorf("failed to create pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+	}
+	defer func() {
+		waitForDeletion(ctx, t, cl, clientPod, 5*time.Minute)
```

Comment:
I think the motivation was to ensure the pod is gone before starting another dig pod query attempt, to avoid a naming collision (since they use the same name). I believe the API delete call isn't blocking.

I've updated to use a random pod name suffix (to ensure no naming collisions), and to call the delete without blocking.